kpack/main.tf

```diff
@@ -0,0 +1,105 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 3.0"
+    }
+    kubectl = {
+      source  = "gavinbunney/kubectl"
+      version = ">= 1.11.0"
+    }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = ">= 2.3.0"
+    }
+  }
+}
+
+provider "kubernetes" {
+  config_path = var.kube_config
+}
+
+resource "kubernetes_namespace" "kpack" {
+  metadata {
+    name = "kpack"
+  }
+}
+
+
+resource "kubernetes_secret" "dockerhub" {
+  depends_on = [kubernetes_namespace.kpack]
+  metadata {
+    name      = "regcreds"
+    namespace = "kpack"
+  }
+  data = {
+    ".dockerconfigjson" = <<DOCKER
+{
+  "auths": {
+    "index.docker.io": {
+      "auth": "${base64encode("${var.dockerhub_user}:${var.dockerhub_password}")}"
+    }
+  }
+}
+DOCKER
+  }
+  type = "kubernetes.io/dockerconfigjson"
+}
+
+data "kubectl_file_documents" "manifests" {
+  #TODO Download the file from github https://github.com/pivotal/kpack/releases/download/v0.3.1/release-0.3.1.yaml
```

Comment:
I would prefer to deploy kpack without having to download anything first. This repo is made of self-contained modules: you don't have to download anything on your workstation.

Reply:
I'll update the PR with the code that downloads it.